src/nvim/api/ui_events.in.h

```diff
@@ -22,7 +22,7 @@ void mouse_on(void)
   FUNC_API_SINCE(3);
 void mouse_off(void)
   FUNC_API_SINCE(3);
-void mode_change(String mode, Integer mode_idx)
+void mode_change(String mode, Integer mode_idx_)
```

Comment:
This name is exported in metadata. Better rename whatever internal variable instead.

Reply:
@justinmk I guess it is `static mode_idx` in `ui.c`, can rename that to `static current_mode_idx`.

Reply:
Yeah, I thought it might be ok since param names aren't part of the API contract. But I will change it.

Reply:
It is ok to change it. But the point of changing an exported name should be to _improve_ the exported interface, not to fix unrelated internal trouble.